app/src/androidTest/java/com/reach5/identity/sdk/demo/SandboxSuccessTest.kt

```diff
@@ -0,0 +1,214 @@
+
+        val textView = onView(
+                allOf(
+                        withId(R.id.user_email),
+                        withParent(withParent(withId(android.R.id.content))),
+                        isDisplayed()
+                )
+        )
+        Thread.sleep(4_000)
+        textView.check(matches(isDisplayed()))
+        textView.check(matches(withText(newEmail)))
+        Espresso.pressBack()
+
+        // login with email
+        val appCompatLoginButton = onView(
+                allOf(
+                        withId(R.id.passwordLogin), withText("Login"),
+                        childAtPosition(
+                                childAtPosition(
+                                        withClassName(`is`("android.widget.LinearLayout")),
+                                        1
+                                ),
+                                0
+                        )
+                )
+        )
+        Thread.sleep(2_000)
+        appCompatLoginButton.perform(scrollTo(), click())
+
+        val textViewLogin = onView(
+                allOf(
+                        withId(R.id.user_email),
+                        withParent(withParent(withId(android.R.id.content))),
+                        isDisplayed()
+                )
+        )
+        Thread.sleep(4_000)
+        textViewLogin.check(matches(isDisplayed()))
+        textViewLogin.check(matches(withText(newEmail)))
+
+        // signup with phone
+        Espresso.pressBack()
+        appCompatEditText.perform(
+                scrollTo(),
+                replaceText(""),
+                closeSoftKeyboard()
+        )
+
+        val appCompatEditTextPhone = onView(
+                allOf(
+                        withId(R.id.phoneNumber),
+                        childAtPosition(
+                                childAtPosition(
+                                        withClassName(`is`("android.widget.LinearLayout")),
+                                        0
+                                ),
+                                3
+                        )
+                )
+        )
+
+        appCompatEditTextPhone.perform(
+                scrollTo(),
+                replaceText(newPhone),
+                closeSoftKeyboard()
+        )
+        Thread.sleep(2_000)
+        appCompatButton.perform(scrollTo(), click())
+
+        val textViewPhone = onView(
+                allOf(
+                        withId(R.id.user_phone_number),
+                        withParent(withParent(withId(android.R.id.content))),
+                        isDisplayed()
+                )
+        )
+        Thread.sleep(4_000)
+        textViewPhone.check(matches(isDisplayed()))
+        textViewPhone.check(matches(withText("+33"+newPhone)))
+
+        // login with phone
+        Thread.sleep(2_000)
+        Espresso.pressBack()
+        appCompatLoginButton.perform(scrollTo(), click())
+
+        Thread.sleep(4_000)
```

Comment:
24 seconds spend to sleep. It is very long.
Can you implement a polling strategy instead ?

Reply:
Ok I will see that